src/main/java/com/hierynomus/spnego/NegTokenInit.java

```diff
@@ -102,11 +103,18 @@ public NegTokenInit read(byte[] bytes) throws IOException {
     public NegTokenInit read(Buffer<?> buffer) throws IOException {
         try {
             ASN1Primitive applicationSpecific = new ASN1InputStream(buffer.asInputStream()).readObject();
-            if (!(applicationSpecific instanceof ASN1ApplicationSpecific)) {
+            if (!(applicationSpecific instanceof BERApplicationSpecific || applicationSpecific instanceof DERApplicationSpecific)) {
```

Comment:
Related to our project where we are unable to move to the new BC because of another dependency. Since there is no change to the functionality, thought it might be ok.